app/components/settings/settings.component.ts

```diff
@@ -97,4 +98,13 @@ export class SettingsComponent implements OnDestroy {
             }
         });
     }
+
+    private _isValid() {
+        try {
+            JSON.parse(stripJsonComments(this.userSettings.value, { whitespace: true }));
+            return true;
+        } catch (e) {
+            return false;
```

Comment:
do we need to report any formatting errors to the user?

Reply:
it already done above, this was just because the ctrl s shortcut could save invalid settings and monaco also show more detailed error now